main.c

```diff
@@ -213,7 +222,13 @@ static void handle_tty(int ev)
 	char data[256];
 	int len;
 
-	if (ev & EPOLLIN) {
+	if (ev & EPOLLHUP) {
```

Comment:
Why were these swapped? We should keep reading while EPOLLIN is set even if the pipe has been closed and this works correctly on linux I believe, is there some difference of behaviour on FreeBSD?

Reply:
It might be a difference between epoll on Linux and our shim implementation. When we receive ctrl+d both flags are set and we don't read anything on the other end of the pipe (i.e. len=0) so we call abort() and core dump. I believe that this code should work on Linux but I'll have a look at our implementation of epoll.

Reply:
Ok, I'd be fine with removing the `assert(len)` call and adding a comment noting the difference of behaviour, but the order of the `if`s can't be swapped, that would be wrong.